modules/bluefish/util/blue_velvet.h

```diff
@@ -33,25 +32,213 @@
+	typedef bool(__cdecl *pFunc_bfcUtilsIsVideoModeProgressive)(const unsigned int nVideoMode);
+	typedef bool(__cdecl *pFunc_bfcUtilsIsVideoMode1001Framerate)(const unsigned int nVideoMode);
+	typedef int(__cdecl *pFunc_bfcUtilsGetFpsForVideoMode)(const unsigned int nVideoMode);
+	typedef int(__cdecl *pFunc_bfcUtilsGetVideoModeForFrameInfo)(const BLUE_UINT32 nWidth, const BLUE_UINT32 nHeight, const BLUE_UINT32 nRate, const BLUE_UINT32 bIs1001, const BLUE_UINT32 bIsProgressive, BLUE_UINT32& nVideoMode);
+	typedef int(__cdecl *pFunc_bfcUtilsGetFrameInfoForVideoMode)(const BLUE_UINT32 nVideoMode, BLUE_UINT32&  nWidth, BLUE_UINT32& nHeight, BLUE_UINT32& nRate, BLUE_UINT32& bIs1001, BLUE_UINT32& bIsProgressive);
+	typedef int(__cdecl *pFunc_bfcUtilsGetAudioSamplesPerFrame)(const BLUE_UINT32 nVideoMode, const BLUE_UINT32 nFrameNo);
+
+
+public:
+
+	BvcWrapper();						// bfcFactory
+	~BvcWrapper();						// bfcDestory
+
+	bool		IsBvcValid();
+	const char*	GetVersion();
+
+	BLUE_UINT32 Enumerate(int& iDevices);
+	BLUE_UINT32 QueryCardType(int& iCardType, int iDeviceID);
+
+	BLUE_UINT32 Attach(int iDeviceId);
+	BLUE_UINT32 Detach();
+
+	BLUE_UINT32 QueryCardProperty32(const int iProperty, unsigned int& nValue);
+	BLUE_UINT32 SetCardProperty32(const int iProperty, const unsigned int nValue);
+	
+	BLUE_UINT32 SystemBufferWrite(unsigned char* pPixels, unsigned long ulSize, unsigned long ulBufferID, unsigned long ulOffset);
+	BLUE_UINT32 SystemBufferRead(unsigned char* pPixels, unsigned long ulSize, unsigned long ulBufferID, unsigned long ulOffset);
+
+	BLUE_UINT32 VideoPlaybackStop(int iWait, int iFlush);
+	BLUE_UINT32 WaitVideoOutputSync(unsigned long ulUpdateType, unsigned long& ulFieldCount);
+	BLUE_UINT32 WaitVideoInputSync(unsigned long ulUpdateType, unsigned long & ulFieldCount);
+
+	BLUE_UINT32 RenderBufferUpdate(unsigned long ulBufferID);
+	BLUE_UINT32 RenderBufferCapture(unsigned long ulBufferID);
+
+	BLUE_UINT32 EncodeHancFrameEx(unsigned int nCardType, struct hanc_stream_info_struct* pHancEncodeInfo, void* pAudioBuffer, unsigned int nAudioChannels, unsigned int nAudioSamples, unsigned int nSampleType, unsigned int nAudioFlags);
+	BLUE_UINT32 DecodeHancFrameEx(unsigned int nCardType, unsigned int* pHancBuffer, struct hanc_decode_struct* pHancDecodeInfo);
+
+	BLUE_UINT32 GetFrameInfoForVideoVode(const unsigned int nVideoMode, unsigned int&  nWidth, unsigned int& nHeight, unsigned int& nRate, unsigned int& bIs1001, unsigned int& bIsProgressive);
+	BLUE_UINT32 GetBytesPerFrame(EVideoMode nVideoMode, EMemoryFormat nMemoryFormat, EUpdateMethod nUpdateMethod, unsigned int& nBytesPerFrame);
+
+private:
+	bool				InitFunctionsPointers();
+	BLUEVELVETC_HANDLE	bvc;
+#if defined(_WIN32)
+	HMODULE				hModule;
+#else
+	void*				hModule;
+#endif
+
+//BlueVelvetC function pointers members
+	pFunc_bfcGetVersion bfcGetVersion;
```

Comment:
All of these function pointers should be initialized to `nullptr`. Especially `bfcFactory`.